markdown/developers/modules.md

```diff
@@ -510,6 +510,14 @@ process {
 
 3. Optional inputs are not currently supported by Nextflow. However, passing an empty list (`[]`) instead of a file as a module parameter can be used to work around this issue.
 
+4. Optional outputs SHOULD be marked as optional:
+
+   ```nextflow
+   tuple val(meta), path('*.tab')                   , optional:true, emit: tab
+   ```
+
+5. Since it is not currently possible to mark individual elements of a tuple as optional, optional outputs SHOULD NOT be placed in output channels with mandatory outputs.
```

Comment:
```suggestion
5. Each output file SHOULD be emitted in it's own channel, along with the `meta` map if provided ( the exception is the versions.yml ).
```

Reply:
@drpatelh @jfy133 This is usually enforced right? Each file should be in it's own channel?

Reply:
Ahh, that's new on me, I have a couple of modules with multiple-file output tuples

Reply:
yes, we enforce that - as we don't want to make assumptions what people want to use each output file from (what you may consider should 'travel together' may not be what others do...). So yes, lots of re-joining tuples...